crates/rpc/rpc-eth-api/src/helpers/call.rs

```diff
@@ -759,7 +763,9 @@ pub trait Call:
             apply_block_overrides(*block_overrides, db, &mut evm_env.block_env);
         }
         if let Some(state_overrides) = overrides.state {
-            apply_state_overrides(state_overrides, db)?;
+            apply_state_overrides(state_overrides, db).map_err(|e| {
+                <EthApiError as From<alloy_evm::overrides::StateOverrideError<_>>>::from(e)
```

Comment:
can we also use Self::Error::from_eth_err here?

Reply:
with the current bounds, rustc can't seem to figure out what `From` impl to use when using `Self::Error::from_eth_err`

It's prioritizing the `From` at https://github.com/paradigmxyz/reth/blob/b1e91ea2698f39de18151f04f9a3ee3860809e92/crates/rpc/rpc-eth-api/src/helpers/call.rs#L736-L739

wdyt?